exec/exec.go

```diff
@@ -76,21 +76,21 @@ func (d *Exec) Run(tableName string, keyName string, command string, sleepStartR
 	logrus.WithFields(logrus.Fields{
 		"command": command,
 	}).Info("Executing command")
-	out, err := runExec(command)
+	out, cmdErr := runExec(command)
```

Comment:
I don't think I do understand idae with `cmdErr`. Would you mind to tell me more about it?

Reply:
As far I as understand, you don't want to fail early with a lock error. In case I am correct with this statement, you should make release lock operation not failing when it tries to release a lock that is not acquired.

Reply:
But anyway, when I think about it I don't think I do understand the context. 😞

Reply:
Adding to this thread, since it relevant: 

> I also updated the Run function to always fallback to returning the
error from the command execution and not exiting early.

Curious why not returning exiting early if `runExec` has failed? Is it so to release the lock? If so should we just release the lock the and return early? Also there are two other return early after that, so can get a bit confusing 👍🏾

Reply:
The way I'm thinking is this:
1. Why should lock-exec care whether or not the execution succeeded/failed?
2. The piece that we do care is that the caller can get the execution status unless the lock operation failed.

Reply:
I think, I understand it now. I am pretty on the same boat. lock-exec is thin execution wrapper that does pre (acquire a lock) and post (release a lock) actions. No matter of commadn execution return code, it should always call these operations. But lock-exec should pass through return code from an command, e.g. when lock operations are successful and command execution fails, lock exec should give back non successful return code to a parent shell.

Reply:
Agree! and def makes sense. Might be worth looking into https://golang.org/pkg/os/exec/#ExitError and https://golang.org/pkg/os/#ProcessState.ExitCode, if we want to deeper integrations/redirection 👍🏾